index.php

```diff
@@ -567,17 +571,31 @@ public function deleteFileThemePluginAction(): void
 			return;
 		}
 		if (isset($_REQUEST['deleteThemePlugin'], $_REQUEST['type']) && $this->verifyFormActions(true)) {
-			$filename = str_ireplace(['/', './', '../', '..', '~', '~/', '\\'], null,
-				trim($_REQUEST['deleteThemePlugin']));
-			$type = $_REQUEST['type'];
+			$allowedDeleteTypes = ['files', 'plugins', 'themes'];
+			$filename = str_ireplace(
```

Comment:
What about filenames that contain `....//`?
Replace `../` in `....//` and you leave `../` behind. Eg:

`....//image.png` would turn into `../image.png`

Maybe shell unicode escape sequences could be used to bypass this protection.

What about filtering filenames and whitelisting characters instead of blacklisting?